composer/loggers/in_memory_logger.py

```diff
@@ -74,3 +77,58 @@ def log_metric(self, timestamp: Timestamp, log_level: LogLevel, data: TLogData):
             self.data[k].append((timestamp, log_level, v))
         self.most_recent_values.update(data)
         self.most_recent_timestamps.update({k: timestamp for k in data})
+
+    def get_timeseries(self, metric: str) -> Dict[str, TLogData]:
+        """Returns logged data as dict containing values of a desired metric over time.
+
+        Args:
+            metric (str): Metric of interest. Must be present in self.data.keys().
+
+        Returns:
+            timeseries (Dict[str, TLogData]): Dictionary in which one key is ``metric``,
+                and the associated value is a list of values of that metric. The remaining
+                keys are each a unit of time, and the associated values are each a list of
+                values of that time unit for the corresponding index of the metric. For
+                example:
+                >>> InMemoryLogger.get_timeseries(metric="accuracy/val")
+                {"accuracy/val": [31.2, 45.6, 59.3, 64.7, "epoch": [1, 2, 3, 4, ...],
+                ...], "batch": [49, 98, 147, 196, ...], ...}
+
+        Example:
+            .. testcode::
+
+                import matplotlib.pyplot as plt
+
+                from composer.core.logging import LogLevel
+                from from composer.loggers import InMemoryLogger
+
+                in_mem_logger = InMemoryLogger(LogLevel.BATCH)
+                in_mem_logger = populate_in_memory_logger(in_mem_logger, datafield="accuracy/val")
```

Comment:
Might be better to directly populate it in the testcode here with some minimal data, e.g.
```
# for this example, directly populate the data
in_mem_logger.data = {'accuracy/val': [(1, blah, 0.4), (2, blah, 1.0)]}
timeseries = in_mem_logger.get_timeseries("accuracy/val")
timeseries
>>> [0.4, 1.0]
```

Reply:
I populate in the example now, but it's pretty verbose (as you can see) because it requires `Timestamp`s. I could go either way!